app/models/cart.rb

```diff
@@ -7,21 +7,34 @@ def initialize(donations={})
 
   def donations
     if @donations
-      @donations.map do |candidate_issue_id, amount|
-        Donation.new(candidate_issue_id: candidate_issue_id, amount: amount)
-      end
+      Donation.transform(@donations)
     end
   end
 
   def create(params)
     candidate_issue_id = params[:donation][:candidate_issue_id]
-    donations ||= {}
-    donations[candidate_issue_id] = (donations[candidate_issue_id].to_i
+    donation ||= {}
+    donation[candidate_issue_id] = (donation[candidate_issue_id].to_i
                                   + params[:donation][:amount].to_i).to_s
-    donations
+    donation
+    validate_session(donation)
   end
 
-  def total
+  def validate_session(donation)
+    if @donations
+      add_to_cart(donation)
+    else
+      donation
+    end
+  end
+
+  def add_to_cart(donation)
+    @donations.merge(donation) do |key, old_val, new_val|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.